es-app/src/views/SystemView.cpp

```diff
@@ -419,6 +428,9 @@ void  SystemView::getViewElements(const std::shared_ptr<ThemeData>& theme)
 	if (sysInfoElem)
 		mSystemInfo.applyTheme(theme, "system", "systemInfo", ThemeFlags::ALL);
 
+	mScrollSound = Sound::getPath(theme, "system", "scroll");
```

Comment:
These seem to be set in applyTheme - is there a case where sysInfoElem doesn't exist yet we'll reload these sounds without applying the theme?

Reply:
Sorry - just re-read: probably it's the other way around - you only want to set these if the new theme is applied, then (meaning, inside the if)?

Reply:
The "if" is only there to make sure the ThemeElement is not null, Sound::getPath already do that check and return an empty string if it happen.

Reply:
I understand that the getPath getter is safe. Still, unless there is a use case for the sound to be updated when the theme is not, this is a case where these should be inside the if.

I take it that mScrollSound and mLaunchSound are initialized somewhere else with default values.

Reply:
I'm not sure what you mean. The 2 "if" that we see before just check if the theme element is here and if it is, it is applied. The "systemInfo" is just a sub element of the system theme, just like the sound.

Reply:
Got it - nevermind.